packages/material-ui/src/Slider/Slider.js

```diff
@@ -358,6 +359,10 @@ const shouldSpreadStyleProps = (Component) => {
 
 const Slider = React.forwardRef(function Slider(inputProps, ref) {
   const props = useThemeProps({ props: inputProps, name: 'MuiSlider' });
+
+  const theme = useTheme();
```

Comment:
Did you consider this API?

```suggestion
  const { props, theme } = useThemeProps({ props: inputProps, name: 'MuiSlider' });
```

Reply:
I think it’s simpler to return just props, the theme is not required most of the time, and it can be access trough it’s own hook.